airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.java

```diff
@@ -83,4 +132,347 @@ protected DataType<?> toDialectType(final AirbyteProtocolType airbyteProtocolTyp
+    metaColumns.put(COLUMN_NAME_AB_EXTRACTED_AT, SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false));
+    if (includeMetaColumn)
+      metaColumns.put(COLUMN_NAME_AB_META, getStructType().nullable(false));
+    return metaColumns;
+  }
+
+  /**
+   * build jooq fields for raw table with type-casted data columns first and then meta columns without
+   * _airbyte_meta.
+   *
+   * @param columns
+   * @param metaColumns
+   * @return
+   */
+  @VisibleForTesting
+  List<Field<?>> buildRawTableSelectFields(final LinkedHashMap<ColumnId, AirbyteType> columns, final Map<String, DataType<?>> metaColumns) {
+    final List<Field<?>> fields =
+        metaColumns.entrySet().stream().map(metaColumn -> field(quotedName(metaColumn.getKey()), metaColumn.getValue())).collect(Collectors.toList());
+    // Use originalName with non-sanitized characters when extracting data from _airbyte_data
+    final List<Field<?>> dataFields = extractRawDataFields(columns);
+    dataFields.addAll(fields);
+    return dataFields;
+  }
+
+  @VisibleForTesting
+  Condition rawTableCondition(DestinationSyncMode syncMode, boolean isCdcDeletedAtPresent, Optional<Instant> minRawTimestamp) {
+    Condition condition = field(name(COLUMN_NAME_AB_LOADED_AT)).isNull();
+    if (syncMode == DestinationSyncMode.APPEND_DEDUP) {
+      if (isCdcDeletedAtPresent) {
+        condition = condition.or(cdcDeletedAtNotNullCondition());
+      }
+    }
+    if (minRawTimestamp.isPresent()) {
+      condition = condition.and(field(name(COLUMN_NAME_AB_EXTRACTED_AT)).gt(minRawTimestamp.get().toString()));
+    }
+    return condition;
+  }
+
+  @Override
+  public String createTable(final StreamConfig stream, final String suffix, final boolean force) {
+    // TODO: Use Naming transformer to sanitize these strings with redshift restrictions.
+    String finalTableIdentifier = stream.id().finalName() + suffix.toLowerCase();
+    if (!force) {
+      return Strings.join(
+          List.of(
+              createSchemaSql(stream.id().finalNamespace()),
+              createTableSql(stream.id().finalNamespace(), finalTableIdentifier, stream.columns())),
+          ";" + System.lineSeparator());
+    }
+    return Strings.join(
+        List.of(
+            createSchemaSql(stream.id().finalNamespace()),
+            beginTransaction(),
+            dropTableIfExists(quotedName(stream.id().finalNamespace(), finalTableIdentifier)),
+            createTableSql(stream.id().finalNamespace(), finalTableIdentifier, stream.columns()),
+            commitTransactionInternal()),
+        ";" + System.lineSeparator());
+  }
+
+  @Override
+  public String updateTable(final StreamConfig streamConfig,
+                            final String finalSuffix,
+                            final Optional<Instant> minRawTimestamp,
+                            final boolean useExpensiveSaferCasting) {
+
+    // TODO: Add flag to use merge vs insert/delete
```

Comment:
I think insert+delete here is fine? iirc (?) MERGE has very specific syntax per destination, IMO we shouldn't even try to genericize it

Reply:
Agreed. Just leaving an abstract method for implementation specific.  fwiw redshift, postgres and SqlServer has same syntax for merge I think.